cloud/ec2.go

```diff
@@ -595,25 +595,21 @@ func (m *ec2Manager) setNoExpiration(ctx context.Context, h *host.Host, noExpira
 	}
 
 	if noExpiration {
-		instance, err := m.client.GetInstanceInfo(ctx, h.Id)
+		if err := h.MarkShouldNotExpire(ctx, expireOnValue); err != nil {
+			return errors.Wrapf(err, "marking host should not expire in DB for host '%s'", h.Id)
+		}
+
+		// Add/update the persistent DNS name if the unexpirable host is
+		// running.
+		_, err := m.GetInstanceStatus(ctx, h)
 		grip.Error(message.WrapError(err, message.Fields{
-			"message":    "could not get instance info for assigning persistent DNS name",
+			"message":    "could not get instance info to assign persistent DNS name",
 			"dashboard":  "evergreen sleep schedule health",
 			"host_id":    h.Id,
 			"started_by": h.StartedBy,
 		}))
-		if instance != nil {
-			grip.Error(message.WrapError(setHostPersistentDNSName(ctx, m.env, h, utility.FromStringPtr(instance.PublicIpAddress), m.client), message.Fields{
-				"message":         "could not update host's persistent DNS name",
-				"op":              "upsert",
-				"dashboard":       "evergreen sleep schedule health",
-				"host_id":         h.Id,
-				"started_by":      h.StartedBy,
-				"instance_status": ec2StatusToEvergreenStatus(instance.State.Name),
-			}))
-		}
```

Comment:
If we still want to add/update the persistent DNS name here if the unexpirable host is running, we would still need this code.

Reply:
`GetInstanceStatus` already updates the host's cached data (including DNS names, IP addresses, tags, etc.). I can mention that it does this internally.